### DISCUSSION TITLE
Camelyon17 Loss Function

### DISCUSSION THREAD

**Discussion by niki-amini-naieni** [p-lambda/wilds#102] [Q&A] [answered]
<img width="634" alt="image" src="https://user-images.githubusercontent.com/61301874/151244957-f02dea77-02a1-401d-9a86-f48a99bd8630.png">
Hello,

Can someone please confirm that this is the loss function (with the same arguments) that was used for the experiments on the Camelyon17 dataset in the original WILDS paper? 

It seems as though to transform the output to a scalar, you eventually compute the mean:
<img width="412" alt="image" src="https://user-images.githubusercontent.com/61301874/151536350-63f7a9ec-405b-4fb1-a25f-1bbe1ac7ffbf.png">

I am wondering why then you set `reduction='none'` instead of leaving `reduction=mean`, which is the default for PyTorch.

Thanks for all the help!

**Comment by teetone** [comment, marked as answer]
Hi @niki-amini-naieni, yes, that's correct.

We don't initialize the loss with `reduction=mean` because we use the unreduced loss to compute group-wise metrics for GroupDRO (see `def _compute_group_wise` under `def _compute`).

**Reply by niki-amini-naieni** [reply]
Thank you very much!
